src/main/kotlin/com/terraformation/backend/tracking/ObservationService.kt

```diff
@@ -232,20 +236,117 @@ class ObservationService(
   ): ReplacementResult {
     requirePermissions { replaceObservationPlot(observationId) }
 
-    val observation = observationStore.fetchObservationById(observationId)
-    val plots = observationStore.fetchObservationPlotDetails(observationId)
-    val plot =
-        plots.firstOrNull { it.model.monitoringPlotId == monitoringPlotId }
-            ?: throw PlotNotInObservationException(observationId, monitoringPlotId)
+    return observationStore.withLockedObservation(observationId) { observation ->
+      val addedPlotIds = mutableSetOf<MonitoringPlotId>()
+      val removedPlotIds = mutableSetOf(monitoringPlotId)
+      val plantingSite =
+          plantingSiteStore.fetchSiteById(observation.plantingSiteId, PlantingSiteDepth.Plot)
+      val plantingZone =
+          plantingSite.plantingZones.first { zone ->
+            zone.plantingSubzones.any { subzone ->
+              subzone.monitoringPlots.any { it.id == monitoringPlotId }
+            }
+          }
+      val plantingSubzone =
+          plantingZone.plantingSubzones.first { subzone ->
+            subzone.monitoringPlots.any { it.id == monitoringPlotId }
+          }
+      val observationPlots = observationStore.fetchObservationPlotDetails(observationId)
+      val observationPlotIds = observationPlots.map { it.model.monitoringPlotId }.toSet()
+      val observationPlot =
+          observationPlots.firstOrNull { it.model.monitoringPlotId == monitoringPlotId }
+              ?: throw PlotNotInObservationException(observationId, monitoringPlotId)
+
+      if (observationPlot.model.completedTime != null) {
+        throw PlotAlreadyCompletedException(monitoringPlotId)
+      }
 
-    if (plot.model.completedTime != null) {
-      throw PlotAlreadyCompletedException(monitoringPlotId)
-    }
+      log.info("Replacing monitoring plot $monitoringPlotId in observation $observationId")
 
-    eventPublisher.publishEvent(
-        ObservationPlotReplacedEvent(duration, justification, observation, monitoringPlotId))
+      if (observationPlot.model.isPermanent) {
+        val isFirstObservation =
+            observationStore.fetchObservationsByPlantingSite(observation.plantingSiteId).none {
+              it.state == ObservationState.Completed
+            }
```

Comment:
In theory there's nothing in the data model to prevent that, but if it happens, there has been a major breakdown on the business side and it won't matter how we handle plot replacement. Probably the right solution is to forbid having multiple in-progress observations (and enforce that rule in the code).